src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/ParameterNumberCheck.java

```diff
@@ -86,6 +94,11 @@ public class ParameterNumberCheck
     /** Ignore number of parameters for methods with {@code @Override} annotation. */
     private boolean ignoreOverriddenMethods;
 
+    /**
+     * Ignore methods and constructors annotated with the specified annotation(s).
+     */
+    private Set<String> ignoreAnnotatedBy = Set.of();
```

Comment:
@nrmancuso @romani We have `ignoreOverriddenMethods` which also ignores methods based on annotation of `Override`. Is there some reason we aren't combining these 2 to make the configuration for this check more reasonable?

Reply:
I was thinking the same, but then I read:
https://github.com/checkstyle/checkstyle/issues/11340#issuecomment-1052311653

Reply:
We could deprecate this property, and remove it it the next major version bump.

Reply:
Do we have a resolution here?

Reply:
We can leave both for now, let’s create an issue to discuss deprecation of ignoreOverriddenMethods

Reply:
https://github.com/checkstyle/checkstyle/issues/14663

Reply:
@rnveach please continue review